### ISSUE TITLE
React: tests fail after chapter 9

### ISSUE THREAD

**Issue by mauritsvanrees** [plone/training#424]
After chapter 9, `yarn test` fails in various ways.

The FaqItem tests need an `index` and `onDelete` argument, something like this:

```
import React from "react";
import renderer from "react-test-renderer";

import FaqItem from "./FaqItem";

describe("FaqItem", () => {
  it("renders a FAQ item", () => {
    function deleter(index) {
      alert(index);
    };
    const component = renderer.create(
      <FaqItem
        question="What is the answer to life the universe and everything?"
        answer="42"
        index={0}
        onDelete={deleter}
      />
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
```

(And the snapshot should be updated, if you have saved it.)

And then we still have a test failure in the App test, same as in issue #423:

      Warning: Each child in a list should have a unique "key" prop.

I don't know how to fix that.

**Comment by ichim-david** [comment]
@mauritsvanrees  I'm glad you took the time to go through the tutorial and add bug reports for
issues that you've found.
I just finished the tutorial last night and while grateful that it was prepared I was surprised of
the following things that I found:
1. the key prop warning was present and never addressed and fixed
2. tests were written at the beginning and then you get failing tests for you have extra required props
3. Express backend is introduced and yet it's just a static array of objects and when you view a newly added faq
  compoentWillMount will remove from display that item
4. Confusing middleware and redux tied with Express however in the end you are not really using the express code
  to persist but only to replace what was added in the state of the components

Hopefully as we gain more knowledge we can also contribute to this lack of information as part of these trainings.
